File: goalbot-backend/src/main/java/com/example/goalbot/entity/Checkin.java
package com.example.goalbot.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("checkin")
public class Checkin {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private Long taskId;

    private Integer actualMinutes;

    private String content;

    private Integer mood;

    private Integer difficulty;

    private LocalDateTime createdAt;
}

File: goalbot-frontend/src/types/checkin.ts
export interface Checkin {
  id: number
  userId: number
  taskId: number
  taskTitle?: string
  actualMinutes: number
  content?: string
  mood?: number
  difficulty?: number
  createdAt: string
}

export interface CheckinPayload {
  taskId: number
  actualMinutes?: number
  content?: string
  mood?: number
  difficulty?: number
}

export interface CheckinStats {
  totalMinutes: number
  checkinCount: number
  completedTaskCount: number
  averageMood: number
  averageDifficulty: number
}
